File: Longest_Subsequence/longest.go
package longestsubsequence

import "sort"

func answerQueries(nums []int, queries []int) []int {
	sort.Ints(nums)
	result := make([]int, len(queries))

	for index, num := range queries {
		sum, subsequenceCount := 0, 0

		for j := 0; j < len(nums); j++ {
			if sum+nums[j] <= num {
				sum += nums[j]
				subsequenceCount++
			} else {
				break
			}
		}

		result[index] = subsequenceCount
	}

	return result
}

File: Longest_Subsequence/longest.py
from typing import List


class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        result: List[int] = [0] * len(queries)
        nums = sorted(nums)

        result_index = 0

        while result_index < len(queries):

            sum, subsequence_count, num_index = 0, 0, 0

            while num_index < len(nums):
                if sum + nums[num_index] <= queries[result_index]:
                    sum += nums[num_index]
                    subsequence_count += 1

                else:
                    break

                num_index += 1

            result[result_index] = subsequence_count

            result_index += 1

        return result
